rewrite-core/src/main/java/org/openrewrite/trait/TypeReference.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.trait;
+
+public interface TypeReference {
+
+    String getName();
```

Comment:
We might have to add that `getTree()` method from `Trait` or some other way so we can correlate it with the LST and later also update the LST.

Reply:
I didn't want to tightly couple `TypeReference` to `Trait` but at that point it might make sense to have `TypeReference` extend `Trait<Tree>` perhaps?